### PR TITLE
added apps-get-logs tool  with comprehensive test coverage

### PR DESCRIPTION
**Pull request by AltuisticIsopod**
- Add getAppLogs function to retrieve DigitalOcean App Platform logs using godo client's **GetLogs**
- Support all log types: BUILD, DEPLOY, RUN, RUN_RESTARTED
- Handle optional parameters (Follow, TailLines) with follow=false and tailLines =100 defaults
- Add comprehensive test suite covering validation, error cases, and edge cases
- Include tests for parameter validation, type checking, and API integration

### CODE REVIEW

**Comment by AltuisticIsopod** [comment]
Hi @ssaengs, Thank you for assigning it to me. Please review my PR and can you tag this issue under Hacktoberfest.

File: internal/apps/apps.go [hunk 1]
```diff
@@ -247,6 +247,67 @@ func (a *AppPlatformTool) updateApp(ctx context.Context, req mcp.CallToolRequest
...
+		return mcp.NewToolResultError("Component is required and must be a string"), nil
+	}
+
+	logTypeStr, ok := req.GetArguments()["LogType"].(string)
+	if !ok {
+		return mcp.NewToolResultError("LogType is required and must be a string"), nil
+	}
+
+	// Validate log type
+	var logType godo.AppLogType
```
Comment (ssaengs) [inline internal/apps/apps.go]: Can we delegate this validation to godo instead? I'm a bit on the fence about adding extra validation for tools that are mapped 1-1 to endpoints.

**Comment by AltuisticIsopod** [comment]
Got it! I’ve delegated the validation to godo by converting the string to godo.AppLogType and letting the API handle invalid values.

**Comment by AltuisticIsopod** [comment]
Hey @ssaengs , Thank you for accepting my PR. Can you please add hacktoberfest-accepted label to my PR?
